app/src/main/cpp/skyline/services/nvdrv/devices/nvhost_as_gpu.cpp

```diff
@@ -107,4 +87,61 @@ namespace skyline::service::nvdrv::device {
 
         state.process->WriteMemory(region, buffer.output.at(0).address);
     }
+
+    void NvHostAsGpu::GetVaRegions(IoctlData &buffer) {
+        struct Data {
+            u64 _pad0_;
+            u32 bufferSize;
+            u32 _pad1_;
+
+            struct {
+                u64 offset;
+                u32 page_size;
+                u32 pad;
+                u64 pages;
+            } regions[2];
+        } regionInfo = state.process->GetReference<Data>(buffer.input.at(0).address);
+        state.process->WriteMemory(regionInfo, buffer.output.at(0).address);
+    }
+
+    void NvHostAsGpu::InitializeEx(IoctlData &buffer) {
+        struct Data {
+            u32 bigPageSize;
+            i32 asFd;
+            u32 flags;
+            u32 reserved;
+            u64 vaRangeStart;
+            u64 vaRangeEnd;
+            u64 vaRangeSplit;
+        } addressSpace = state.process->GetObject<Data>(buffer.input.at(0).address);
+    }
+
+    void NvHostAsGpu::Remap(IoctlData &buffer) {
+        struct Entry {
+            u16 flags;
+            u16 kind;
+            u32 nvmapHandle;
+            u32 mapOffset;
+            u32 gpuOffset;
+            u32 pages;
+        };
+
+        size_t entryCount{buffer.input.at(0).size / sizeof(Entry)};
+        std::span entries(state.process->GetPointer<Entry>(buffer.input.at(0).address), entryCount);
+
+        for (auto entry : entries) {
+            try {
+                auto nvmap = state.os->serviceManager.GetService<nvdrv::INvDrvServices>(Service::nvdrv_INvDrvServices)->GetDevice<nvdrv::device::NvMap>(nvdrv::device::NvDeviceType::nvmap)->handleTable.at(entry.nvmapHandle);
+
+                u64 mapAddress = static_cast<u64>(entry.gpuOffset) << 0x10;
+                u64 mapPhysicalAddress = nvmap->address + (static_cast<u64>(entry.mapOffset) << 0x10);
+                u64 mapSize = static_cast<u64>(entry.pages) << 0x10;
```

Comment:
`constexpr u32 ParamShift{0x10}`
Or smth?

Reply:
Do you really not know what the shift is doing ? If so communicate that to the reader `constexpr u32 ArgShift{0x10}; // This shift is seemingly arbitrary`.

Reply:
I don't get you, it's just an arbitrary shift to save space in the ioctl params I presume

Reply:
It isn't arbitrary, it will have a reason behind it in this case it's likely because of the addresses and size having a minimum alignment requirement. Name it something like `MinAlignmentShift`.